src/prefect/tasks/azure/blobstorage.py

```diff
@@ -75,6 +75,12 @@ class BlobStorageUpload(Task):
         - azure_credentials_secret (str, optional): the name of the Prefect Secret
             that stores your Azure credentials; this Secret must be an Azure connection string
         - container (str, optional): the name of the Azure Blob Storage to upload to
+        - overwrite_blob (bool, optional): Whether the blob to be uploaded should overwrite
+                    the current data. If True, upload_blob will overwrite the existing data.
+                    Note for Append blob types: if set to False and the data already exists,
+                    an error will not be raised and the data will be appended to the existing
+                    blob; if set overwrite=True, then the existing append blob will be deleted,
+                    and a new one created.
```

Comment:
```suggestion
                    If the blob is an 'append' type blob and this is `True`, it will be
                    deleted instead of appended.
```
And the same below. I don't think that we need to explain that it will append as normal if this is not set.